pyAudioAnalysis/audioSegmentation2.py

```diff
@@ -536,17 +541,19 @@ class ID of the i-th segment
     accuracy = 0.0
     class_names = []
     cm = np.array([])
-    if not os.path.isfile(model_name):
-        print("mtFileClassificationError: input model_type not found!")
-        return labels, class_names, accuracy, cm
+
+    if mf_object is None and mfm_object is None:
+        if not os.path.isfile(model_name):
+            print("mtFileClassificationError: input model_type not found!")
+            return labels, class_names, accuracy, cm
 
     # Load classifier:
     if model_type == "knn":
         classifier, mean, std, class_names, mt_win, mid_step, st_win, \
-         st_step, compute_beat = at.load_model_knn(model_name)
+         st_step, compute_beat = at.load_model_knn(model_name, False, mf_object)
```

Comment:
Super-minor one:

I'm not very familiar with Python, but maybe you don't have to specify defaults if you use named arguments?

Smth like this: https://treyhunner.com/2018/04/keyword-arguments-in-python/

```python
st_step, compute_beat = at.load_model_knn(model_name, mf_object=mf_object)
```

(here and elsewhere in the diff)